### PR TITLE
Remove static as requirement

### LINKED ISSUE #24496: Says method must be static but instance method shown..
It says as a constraint that:

> Logging methods must be static, partial, and return void.

But the same page shows that you can use instance level method not just static methods so I am confused.

---
#### Document details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 6fc4ecf0-19e4-54d3-c34e-b6eadb80262b
* Version Independent ID: decc74aa-a2d5-8727-f22b-3862bacde737
* Content: [Compile-time logging source generation](https://docs.microsoft.com/en-gb/dotnet/core/extensions/logger-message-generator)
* Content Source: [docs/core/extensions/logger-message-generator.md](https://github.com/dotnet/docs/blob/main/docs/core/extensions/logger-message-generator.md)
* Product: **dotnet-fundamentals**
* GitHub Login: @maryamariyan
* Microsoft Alias: **maariyan**

### LINKED ISSUE #28441: Constraints not correct for instance-based logging
Log method constraints indicate "Logging methods must be static, partial, and return void." but that's not the case for instance-based logging, documented in the same page.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 6fc4ecf0-19e4-54d3-c34e-b6eadb80262b
* Version Independent ID: decc74aa-a2d5-8727-f22b-3862bacde737
* Content: [Compile-time logging source generation](https://docs.microsoft.com/en-us/dotnet/core/extensions/logger-message-generator)
* Content Source: [docs/core/extensions/logger-message-generator.md](https://github.com/dotnet/docs/blob/main/docs/core/extensions/logger-message-generator.md)
* Product: **dotnet-fundamentals**
* GitHub Login: @maryamariyan
* Microsoft Alias: **maariyan**

### PR DESCRIPTION
**Pull request by gewarren**
Fixes #24496, fixes #28441.

### CODE REVIEW

File: docs/core/extensions/logger-message-generator.md [hunk 1]
```diff
@@ -111,11 +111,12 @@ Consider the example logging output when using the `JsonConsole` formatter.
 
 When using the `LoggerMessageAttribute` on logging methods, there are some constraints that must be followed:
 
-- Logging methods must be `static`, `partial`, and return `void`.
+- Logging methods must be `partial` and return `void`.
 - Logging method names must *not* start with an underscore.
 - Parameter names of logging methods must *not* start with an underscore.
 - Logging methods may *not* be defined in a nested type.
 - Logging methods *cannot* be generic.
+- If a logging method is `static`, the `ILogger` instance is required as a parameter.
```
Comment (Youssef1313) [inline docs/core/extensions/logger-message-generator.md:119]: Doc comments on the attribute just mentions `ILogger` must be a parameter. It doesn't say that it's applicable only to static methods

https://github.com/dotnet/runtime/blob/2330b4bef645bf8bcd3011f35968e55192c697f5/src/libraries/Microsoft.Extensions.Logging.Abstractions/src/LoggerMessageAttribute.cs#L16

Is the doc comment outdated? or is it a general requirement regardless of the static-ness of the method?

Comment (Youssef1313) [reply docs/core/extensions/logger-message-generator.md:119]: The doc comment also mentions a `LogLevel` parameter as a requirement.

File: docs/core/extensions/logger-message-generator.md [hunk 2]
```diff
@@ -111,11 +111,12 @@ Consider the example logging output when using the `JsonConsole` formatter.
 
 When using the `LoggerMessageAttribute` on logging methods, there are some constraints that must be followed:
 
-- Logging methods must be `static`, `partial`, and return `void`.
+- Logging methods must be `partial` and return `void`.
 - Logging method names must *not* start with an underscore.
```
Comment (Youssef1313) [inline docs/core/extensions/logger-message-generator.md:115]: It seems that also parameters can't start with an underscore.

https://github.com/dotnet/runtime/blob/2330b4bef645bf8bcd3011f35968e55192c697f5/src/libraries/Microsoft.Extensions.Logging.Abstractions/gen/LoggerMessageGenerator.Parser.cs#L387-L392

File: docs/core/extensions/logger-message-generator.md (hunk 1 above)
Comment (Youssef1313) [reply docs/core/extensions/logger-message-generator.md:119]: Reading through the code. It seems the doc comment is outdated or incorrect.

https://github.com/dotnet/runtime/blob/2330b4bef645bf8bcd3011f35968e55192c697f5/src/libraries/Microsoft.Extensions.Logging.Abstractions/gen/LoggerMessageGenerator.Parser.cs#L403-L411

File: docs/core/extensions/logger-message-generator.md (hunk 2 above)
Comment (gewarren) [reply docs/core/extensions/logger-message-generator.md:115]: That's on the next line :)

**Comment by gewarren** [comment]
Ping for review.

**Review by tdykstra — APPROVED** [review]
LGTM

**Comment by gewarren** [comment]
Thanks Tom!

**Pull request merged**